Sources/Bob/APIs/GitHub/GitHub+BranchChecking.swift

```diff
@@ -21,39 +21,33 @@ import Foundation
 
 public extension GitHub {
     
-    public func branchExists(_ branch: BranchName, success: @escaping (_ branchExists: Bool, _ possibleMatches: [Branch]) -> Void, failure: @escaping (_ error: Error) -> Void) {
-        // Not using GET branch to avoid 404 caused by lacking permissions
-        self.existingBranches(success: { (branches) in
-            var branchExists = false
-            let possibleMatches = branches.filter({ (remoteBranch) -> Bool in
-                if remoteBranch.name == branch.name {
-                    branchExists = true
-                } else {
-                    let distance = remoteBranch.name.levenshtein(to: branch.name)
-                    if Double(distance)/Double(branch.name.characters.count) < 0.25 {
-                        return true
-                    }
+    public func branchExists(_ branch: BranchName) throws -> (branchExists: Bool, possibleMatches: [Branch]) {
```

Comment:
I know its not in the scope of this PR but it would be much nicer if there was an enum eg. : 
```
enum BranchSearchResult {
  case exists
  case doesntExist(posibbleMatches: [Branch])
}
```

Reply:
Agreed. I didn't want to change too many things that aren't directly related to networking. 
This could be the perfect opportunity for you first contribution to `bob` ;)